lib/datasource/hex/index.ts

```diff
@@ -32,12 +33,17 @@ export class HexDatasource extends Datasource {
     // hexPackageName:organizationName
     // hexPackageName is used to pass it in hex dep url
     // organizationName is used for accessing to private deps
-    const hexPackageName = lookupName.split(':')[0];
-    const hexUrl = `${registryUrl}api/packages/${hexPackageName}`;
+    const [hexPackageName, organizationName] = lookupName.split(':');
+    const organizationUrlPrefix = organizationName
+      ? `repos/${organizationName}/`
+      : '';
+    const hexUrl = `${registryUrl}api/${organizationUrlPrefix}packages/${hexPackageName}`;
+    const { token } = hostRules.find({ url: hexUrl });
+    const httpOptions = token ? { headers: { authorization: token } } : {};
```

Comment:
This should already be done in http layer. What's the benefit adding it here?

Reply:
Unfortunately [`applyAuthorization`](https://github.com/renovatebot/renovate/blob/main/lib/util/http/auth.ts#L10) seems to add `Bearer` before the token in the `Authorization` header - this doesn't work with Hex as [it expects the header as `Authorization: token`](https://hexpm.docs.apiary.io/#introduction/authentication). Is there anything else I could use to get the same header value?

Reply:
Shall we add a new authType value, such as `Token-Only`? Could potentially be useful elsewhere for hacking a custom `Authorization` header

Reply:
Reading the authType docs, I would assume it's what ends up in the `Authorization: <type> <credentials>` header template. According to [RFC 7235](https://datatracker.ietf.org/doc/html/rfc7235#appendix-C), the type is required - Hex is simply not compliant with standards - so I wouldn't introduce this as a re-usable option in the HTTP client, unless there are other cases like this.

This is my opinion though, if you'd rather introduce a `Token-Only` authType I can modify the `http` module.

Reply:
I'll like to see this in http layer, even if it's currently only used by hex. So we do not have to much duplicate code here

Reply:
@viceice I went with @rarkins's suggestion and updated the documentation I found that referred to `authType` - let me know if that's the way to go.